RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelRawToClusterCUDA.cc

```diff
@@ -83,19 +86,29 @@ class SiPixelRawToClusterCUDA : public edm::stream::EDProducer<edm::ExternalWork
 };
 
 SiPixelRawToClusterCUDA::SiPixelRawToClusterCUDA(const edm::ParameterSet& iConfig)
-    : rawGetToken_(consumes<FEDRawDataCollection>(iConfig.getParameter<edm::InputTag>("InputLabel"))),
-      digiPutToken_(produces<cms::cuda::Product<SiPixelDigisCUDA>>()),
+    : digiPutToken_(produces<cms::cuda::Product<SiPixelDigisCUDA>>()),
       clusterPutToken_(produces<cms::cuda::Product<SiPixelClustersCUDA>>()),
-      gpuMapToken_(esConsumes<SiPixelROCsStatusAndMappingWrapper, CkfComponentsRecord>()),
-      gainsToken_(esConsumes<SiPixelGainCalibrationForHLTGPU, SiPixelGainCalibrationForHLTGPURcd>()),
-      cablingMapToken_(esConsumes<SiPixelFedCablingMap, SiPixelFedCablingMapRcd>(
-          edm::ESInputTag("", iConfig.getParameter<std::string>("CablingMapLabel")))),
+      isUpgrade_(iConfig.getParameter<bool>("isUpgrade")),
       isRun2_(iConfig.getParameter<bool>("isRun2")),
       includeErrors_(iConfig.getParameter<bool>("IncludeErrors")),
       useQuality_(iConfig.getParameter<bool>("UseQualityInfo")),
       maxFedWords_(iConfig.getParameter<uint32_t>("MaxFEDWords")),
       clusterThresholds_{iConfig.getParameter<int32_t>("clusterThreshold_layer1"),
                          iConfig.getParameter<int32_t>("clusterThreshold_otherLayers")} {
+  if (isRun2_ and isUpgrade_) {
```

Comment:
I'm a bit concerned these flags may become confusing. If I understood correctly, their meaning is
* `isRun2 == true && isUpgrade == false`: Run 2
* `isRun2 == false && isUpgrade == false`: Run 3
* `isRun2 == false && isUpgrade == true`: Run 4 and beyond

How about making a single string parameter (and internal `enum class`) that specifies the exact running period?

On the other hand, the `isUpgrade` appears to switch most of the code of the EDProducer, at which point a completely separate EDProducer could be worth of considering too.

Reply:
> How about making a single string parameter (and internal enum class) that specifies the exact running period?

Make much more sense.

> On the other hand, the isUpgrade appears to switch most of the code of the EDProducer, at which point a completely separate EDProducer could be worth of considering too.

Right, this was my first implementation, to have a different producer to be run for Digi to Clusters production. Was not sure if it was better to have an extra producer or to have a switch for this. I can roll back to the former.